harness/determined/core/_checkpoint.py

```diff
@@ -34,6 +34,107 @@ class DownloadMode(enum.Enum):
     NoSharedDownload = "NO_SHARED_DOWNLOAD"
 
 
+def merge_metadata(all_metadata: List[Dict[str, Any]]) -> Tuple[Dict[str, Any], Dict[str, List]]:
+    """
+      Given a list of metadata, return:
+        - merged metadata
+        - a dict mapping conflicting keys to ranks that would upload them
+
+    Merging scenarios:
+    #  Metadata 1        | Metadata 2          | Conflict |  Result
+    # -------------------|---------------------|----------|-------------------------------
+    #  key1: val         | key2: val           | no       | {key1: val, key2: val}
+    #  key1: val         | key1: val           | yes      | n/a
+    #  key1: [val]       | key1: [val]         | no       | {key1: [val, val]}
+    #  key1: [val]       | key1: val           | yes      | n/a
+    #  key1: [val]       | key1: {key3: val}   | yes      | n/a
+    #  key1: {key3: val} | key1: {key4: val}   | no       | key1: {key3: val, key4: val}
+    #  key1: {key3: val} | key1: val           | yes      | n/a
+    #  key1: {key3: val} | key1: [val]         | yes      | n/a
+    """
+    merged: Dict[str, Any] = {}
+    conflicts: Dict[str, List] = {}
+    for rank, rank_metadata in enumerate(all_metadata):
+
+        for _key in rank_metadata:
```

Comment:
Why the underscore?

Reply:
I thought `key` was a restricted word... well, it's not. removing underscore.